### ISSUE TITLE
Instant crash when running 1.9.5 on Windows

### ISSUE THREAD

**Issue by Stonehawk** [Querz/mcaselector#89]
✅ Minecraft Java Edition
✅ Brain
✅ Computer

1. Downloaded mcaselector-1.9.5.jar

2. Double-clicked mcaselector-1.9.5.jar

**Expected Result**: mcaselector window opens.

**Actual Result**: A black window opens for a split second and immediately disappears,

3. Reviewed instructions:
>if not, you can try `java -jar mcaselector-1.9.5.jar` from your console.

4. Opened cmd from Run prompt

5. Navigated to the directory containing mcaselector-1.9.5.jar
`D:\Minecraft`

6. Typed java -jar mcaselector-1.9.5.jar

**Expected result**: Unknown; just following instructions

**Actual result**:
```
D:\Minecraft>java -jar mcaselector-1.9.5.jar
Exception in thread "main" java.lang.ExceptionInInitializerError
        at net.querz.mcaselector.debug.Debug.dumpf(Debug.java:47)
        at net.querz.mcaselector.Main.main(Main.java:14)
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
        at java.lang.String.substring(Unknown Source)
        at net.querz.mcaselector.Config.<clinit>(Config.java:25)
        ... 2 more
```

7. Give up?

8. NO! Read the requirements again!

✅ Minecraft Installed

9. Cross reference the instructions...

> Minecraft Java Edition comes with a JRE that you can use to start the MCA Selector, so there is no need to install another version of java on your system. On Windows, that java version is usually located in `C:\Program Files (x86)\Minecraft\runtime\jre-x64\bin\`

10. Feel a hunch, and check `C:\Program Files (x86)\Minecraft\runtime\jre-x64\bin\`

11. ... it's not there

12. **IT'S NOT THERE!**

13. Install Java from scratch to make sure Java exists at all.

14. Make sure java is even working
```
D:\Minecraft>java -version
java version "1.8.0_241"
Java(TM) SE Runtime Environment (build 1.8.0_241-b07)
Java HotSpot(TM) 64-Bit Server VM (build 25.241-b07, mixed mode)
```

15. Run Minecraft Java Edition again just to make sure I wasn't hallucinating the fact that I have a real, factual, functional copy of the game, and NOT Bedrock.

16. Double check that I do indeed ACTUALLY have a computer.
```
Windows 10 Pro
Processor:		Intel(R) Core(TM) i5-6600 CPU @ 3.30GHz 3.31GHz
Installed memory (RAM):	16.0 GB
System Type:		64-bit Operating System, x64-based processor
```

17. Checklist items 1 and 3 clearly are satisfied... 

18. Update requirements checklist:
✅ Minecraft Java Edition
❌ Brain
✅ Computer

I hope someday you will find some way of expanding support to brainless users like me, who make a good faith effort to follow the instructions but still can't seem to get anywhere. Thank you anyway, I understand this is very hard and underappreciated work that you're doing and the minecraft community is incredibly lucky that someone is trying to make anything like this at all. 

Unless there's something here that isn't attributed solely to my gross incompetence, I'll keep watching and see if the next update magically works, but for now I'm about ready to just throw up my hands, dismiss any notion that I ever knew anything about computers, and admit that this shit is all way over my head and I am doing nothing more than blindly mimicking the motions and mannerisms of someone competent like I'm a cargo cult,

**Comment by IncredibleHolg** [comment]
If you look into the code in Config.java it  you get

```java
	public static final File DEFAULT_BASE_DIR;

	static {
		// find jar file directory
		String jarPath;
		String path = ".";
		try {
			jarPath = Config.class.getProtectionDomain().getCodeSource().getLocation().toURI().getPath();
			if (!jarPath.endsWith(".jar")) { // no jar
				path = jarPath.replaceAll("build/classes/java/main/$", "");
			} else {
				path = jarPath.substring(0, jarPath.lastIndexOf(File.separator));
			}
		} catch (URISyntaxException ex) {
			ex.printStackTrace();
		}
		DEFAULT_BASE_DIR = new File(path, "cache");
	}
```

Line 25 is ```path = jarPath.substring(0, jarPath.lastIndexOf(File.separator));```
So it fails to get the directory location of your JAR file. 

Try to generate another direcory level and put your JAR e.g. into ``D:\Minecraft\Mcaselector\`` and try again. ( I am currently under Linux where the error does not occur, so I cannot test this )

If this works, its a bug when putting the jar into a top level directory.

**Comment by Querz** [comment]
Sorry, apparently I did not apply a brain when I released 1.9.5. This issue has been fixed in the [latest release](https://github.com/Querz/mcaselector/releases/latest).
